File: src/main/java/net/serebryansky/routing/controller/DurationController.java
package net.serebryansky.routing.controller;

import lombok.extern.slf4j.Slf4j;
import lombok.val;
import net.serebryansky.place.model.Place;
import net.serebryansky.routing.model.DurationMatrixResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

import java.util.List;
import java.util.stream.Collectors;

import static net.serebryansky.common.util.LoggingUtil.logOnEach;
import static org.springframework.http.ResponseEntity.ok;

@RestController
@RequestMapping("durations")
@Slf4j
public class DurationController {
    @PostMapping("matrix")
    public ResponseEntity<Mono<DurationMatrixResponse>> matrix(@RequestBody List<Place> places) {
        val response = new DurationMatrixResponse();
        response.setMatrix(places
                .stream()
                .map(placeStart -> places
                        .stream()
                        .map(placeEnd -> 10.0)
                        .collect(Collectors.toList()))
                .collect(Collectors.toList()));
        return ok(Mono.empty()
                .doOnEach(logOnEach(() -> log.info("Duration matrix for {}", places)))
                .then(Mono.just(response)));
    }
}

File: src/main/kotlin/net/serebryansky/routing/controller/DurationController.kt
package net.serebryansky.routing.controller

import mu.KotlinLogging
import net.serebryansky.place.model.Place
import net.serebryansky.routing.model.DurationMatrixResponse
import org.springframework.http.ResponseEntity
import org.springframework.http.ResponseEntity.ok
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("durations")
class DurationController {

    private val log = KotlinLogging.logger {}

    @PostMapping("matrix")
    suspend fun matrix(@RequestBody places: List<Place>): ResponseEntity<DurationMatrixResponse> {
        log.info { "Duration matrix for $places" }
        val response = DurationMatrixResponse(
            matrix = places.map { places.map { 10.0 } }
        )
        return ok(response)
    }
}
